src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -1290,7 +1290,7 @@ private void ThrowOnInvalidHandle(SafeWinHttpHandle handle)
         
         private Task<bool> InternalSendRequestAsync(WinHttpRequestState state)
         {
-            state.TcsSendRequest = new TaskCompletionSource<bool>();
+            state.TcsSendRequest = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
```

Comment:
/cc: @SidharthNabar 

Switching to async continuations will have perf impact. We could think of a knob (i.e. API addition) similar to WinHTTP's where the user can switch it on and guarantee no sync/blocking during the I/Os.